crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -2075,6 +2112,25 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Validate that the given expression is a valid assignment target.
+    ///
+    /// If the expression is a list or tuple, then validate each element in the list.
+    /// If it's a starred expression, then validate the value of the starred expression.
+    ///
+    /// Report an error for each invalid assignment expression found.
+    pub(super) fn validate_assignment_target(&mut self, expr: &Expr) {
+        match expr {
+            Expr::Starred(ast::ExprStarred { value, .. }) => self.validate_assignment_target(value),
+            Expr::List(ast::ExprList { elts, .. }) | Expr::Tuple(ast::ExprTuple { elts, .. }) => {
+                for expr in elts {
+                    self.validate_assignment_target(expr);
+                }
+            }
+            Expr::Name(_) | Expr::Attribute(_) | Expr::Subscript(_) => {}
+            _ => self.add_error(ParseErrorType::InvalidAssignmentTarget, expr.range()),
+        }
+    }
```

Comment:
This now matches the behavior of CPython.

Earlier, for list, tuples and starred expression, we'd highlight the entire expression as being considered invalid for an assignment target which is incorrect. A list is allowed as an assignment target but it's one or more of the elements inside the list which is an invalid target. Same goes for tuples and starred expression.

Moving the logic in the `Parser` struct also avoids allocating a vector of `ParseError` and instead we can just add it directly.